node/src/main/kotlin/net/corda/node/services/identity/InMemoryIdentityService.kt

```
@@ -67,7 +67,8 @@ class InMemoryIdentityService(identities: Iterable<PartyAndCertificate> = emptyS
                 val space = (0 until index).joinToString("") { "   " }
                 log.error("$space${certificate.toX509CertHolder().subject}")
             }
-            throw e
+            // Log the error and continue.
+            return null
```

Is that right?! Surely it's better to propagate the exception here?

---

yes thats not right, I should handle the exception in the network map cache instead.